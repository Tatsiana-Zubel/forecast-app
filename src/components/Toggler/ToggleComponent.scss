.toggle {
  display:flex;
  justify-content: center;
  margin: 5px 0;

  button {
    width: 50%;
  }
  button:hover{
    cursor: pointer;
  }
  span {
    margin: 0 10px;
  }
}
