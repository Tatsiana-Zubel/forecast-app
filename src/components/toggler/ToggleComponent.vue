<template>
  <div class="toggle">
    <button @click="switchToCelsius()">&#8451;</button>
    <span>|</span>
    <button @click="switchToFahrenheit()">&#8457;</button>
  </div>
</template>

<script>
import { ToggleValues } from '@/shared/ToggleValues'

export default {
  name: "ToggleComponent",
  data() {
    return {
      toggleValue: null
    }
  },
  methods: {
    switchToCelsius() {
      this.toggleValue = ToggleValues.Celsius;
      this.$emit('toggle', this.toggleValue)
    },
    switchToFahrenheit() {
      this.toggleValue = ToggleValues.Fahrenheit
      this.$emit('toggle', this.toggleValue)
    }
  }
}
</script>

<style lang="scss">
.toggle {
  display: flex;
  justify-content: center;
  margin: 5px 0;

  button {
    width: 50%;
  }

  button:hover {
    cursor: pointer;
  }

  span {
    margin: 0 10px;
  }
}
</style>
