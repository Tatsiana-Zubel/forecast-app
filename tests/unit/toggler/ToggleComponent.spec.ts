import { shallowMount } from '@vue/test-utils'
import ToggleComponent from '@/components/toggler/ToggleComponent.vue'

describe('ToggleComponent.vue', () => {
  const wrapper = shallowMount(ToggleComponent)

  // beforeEach(() => {
  //   wrapper.destroy();
  // });

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('has "toggle" class', () => {
    const wrapperClass = wrapper.find('.wrapper')
    expect(wrapperClass.exists())
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has two buttons', () => {
    const buttons = wrapper.findAll('button')
    expect(buttons.length).toBe(2)
  })

  it('should call "switchToCelsius" method', async () => {
    const spy = jest.spyOn(wrapper.vm, 'switchToCelsius')
    const celsiusBtn = wrapper.findAll('button').at(0)
    // @ts-ignore
    await celsiusBtn.trigger('click')
    expect(spy).toHaveBeenCalled()
  })

  it('should emit "toggle" event with toggleValue equal to celsius ', async () => {
    const celsiusBtn = wrapper.findAll('button').at(0)
    // @ts-ignore
    await celsiusBtn.trigger('click')

    expect(wrapper.emitted().toggle).toBeTruthy()
    expect(wrapper.emitted().toggle.length).toBe(1)
    expect(wrapper.emitted().toggle[0]).toEqual(['CELSIUS'])
  })

  it('should call "switchToFahrenheit" method', async () => {
    const spy = jest.spyOn(wrapper.vm, 'switchToFahrenheit')
    const celsiusBtn = wrapper.findAll('button').at(1)
    // @ts-ignore
    await celsiusBtn.trigger('click')
    expect(spy).toHaveBeenCalled()
  })

  it('should emit "toggle" event with toggleValue equal to Fahrenheit ', async () => {
    const celsiusBtn = wrapper.findAll('button').at(1)
    // @ts-ignore
    await celsiusBtn.trigger('click')

    expect(wrapper.emitted().toggle).toBeTruthy()
    expect(wrapper.emitted().toggle.length).toBe(1)
    expect(wrapper.emitted().toggle[0]).toEqual(['FAHRENHEIT'])
  })
})
