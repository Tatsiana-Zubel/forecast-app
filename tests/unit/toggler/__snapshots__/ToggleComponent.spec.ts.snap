// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleComponent.vue should match snapshot 1`] = `
<div
  class="toggle"
>
  <button>
    ℃
  </button>
  <span>
    |
  </span>
  <button>
    ℉
  </button>
</div>
`;
